editing/edit-with-branch-versioning/src/main/java/com/esri/samples/edit_with_branch_versioning/EditWithBranchVersioningController.java

```diff
@@ -0,0 +1,356 @@
+import com.esri.arcgisruntime.data.ServiceGeodatabase;
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.layers.FeatureLayer;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.GeoElement;
+import com.esri.arcgisruntime.mapping.Viewpoint;
+import com.esri.arcgisruntime.mapping.view.IdentifyLayerResult;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.security.AuthenticationManager;
+import com.esri.arcgisruntime.security.DefaultAuthenticationChallengeHandler;
+
+public class EditWithBranchVersioningController {
+
+  @FXML private Button createVersionButton;
+  @FXML private Button switchVersionButton;
+  @FXML private ComboBox<String> damageTypeComboBox;
+  @FXML private ComboBox<VersionAccess> accessComboBox;
+  @FXML private Label currentVersionLabel;
+  @FXML private MapView mapView;
+  @FXML private ProgressIndicator progressIndicator;
+  @FXML private TextField descriptionTextField;
+  @FXML private TextField nameTextField;
+  @FXML private VBox createVersionVBox;
+  @FXML private VBox editFeatureVBox;
+
+  private ArcGISFeature selectedFeature;
+  private FeatureLayer featureLayer;
+  private ServiceFeatureTable serviceFeatureTable;
+  private ServiceGeodatabase serviceGeodatabase; // keeps loadable in scope to avoid garbage collection
+  private String defaultVersion;
+  private String userCreatedVersion;
+
+  public void initialize() {
+    try {
+      // create a map with the streets vector basemap
+      ArcGISMap map = new ArcGISMap(Basemap.createStreetsVector());
+
+      // create a map view and set its map
+      mapView.setMap(map);
+
+      // add the version access types to the combo box
+      accessComboBox.getItems().addAll(VersionAccess.PUBLIC, VersionAccess.PROTECTED, VersionAccess.PRIVATE);
+
+      // add the damage types to the combo box and handle selection
+      damageTypeComboBox.getItems().addAll("Destroyed", "Inaccessible", "Major", "Minor", "Affected");
+      damageTypeComboBox.getSelectionModel().selectedItemProperty().addListener((o, p, n) -> {
+        if (!selectedFeature.getAttributes().get("TYPDAMAGE").equals(n)) {
+          try {
+            selectedFeature.getAttributes().put("TYPDAMAGE", damageTypeComboBox.getValue());
+            updateFeature(selectedFeature);
+          } catch (Exception e) {
+            new Alert(Alert.AlertType.ERROR, "Feature attributes failed to update").show();
+          }
+        }
+      });
+
+      // handle authentication for the service geodatabase
+      AuthenticationManager.setAuthenticationChallengeHandler(new DefaultAuthenticationChallengeHandler());
+
+      // create and load a service geodatabase
+      serviceGeodatabase = new ServiceGeodatabase("https://sampleserver7.arcgisonline" +
+        ".com/arcgis/rest/services/DamageAssessment/FeatureServer");
```

Comment:
I still think this could go in one line (unless we're going to start being more strict about the line character limit)
```suggestion
      serviceGeodatabase = new ServiceGeodatabase("https://sampleserver7.arcgisonline.com/arcgis/rest/services/DamageAssessment/FeatureServer");
```

Reply:
Yeah I think we're trying to be really strict on this. Again, will note this in the contrib guidelines.